pages/categorization.vue

```diff
@@ -1,319 +1,245 @@
 
-					// 1. Produce an array of dicts
-					this.columnToCategoryTable = [];
+                        // I. Save a lowercase version of the current json key
+                        const columnLowercase = column.toLowerCase();
 
-					// A. Each dict has a header entry from the data table file
-					let headerFields = [];
-					for ( let headerField in this.dataTable.original[0] ) {
+                        // II. Try to match the json key with one in the tsv file
+                        if ( headerFields.includes(columnLowercase) ) {
 
-						// I. Save the header field in a list
-						headerFields.push(headerField);
+                            for ( let index = 0; index < this.columnToCategoryTable.length; index++ ) {
 
-						// II. Save a new dict for this column and description
-						this.columnToCategoryTable.push({
+                                // NOTE: Advanced column name matching here between tsv and json? J. Armoza 01/26/22
+                                if ( columnLowercase === this.columnToCategoryTable[index].column.toLowerCase() ) {
 
-							"category": null,
-							"column": headerField,
-							"description": ""
-						});
-					}
+                                    // a. Determine the description string for this json file column entry
+                                    let descriptionStr = "";
+                                    for ( const subkey in this.dataDictionary.original[column] ) {
 
-					// B. and a corresponding "description" column that is (possibly) sourced from the json file
-					for ( let column in this.dataDictionary.original ) {
+                                        if ( "description" === subkey.toLowerCase() ) {
+                                            descriptionStr = this.dataDictionary.original[column][subkey];
+                                            break;
+                                        }
+                                    }    
+                                
+                                    // b. Save the description from the json file colum entry
+                                    this.columnToCategoryTable[index].description = descriptionStr;
+                                }
+                            }
+                        }
+                    }
+                }
+            },            
 
-						// I. Save a lowercase version of the current json key
-						let columnLowercase = column.toLowerCase();
+            tableClick(p_clickData) {
 
-						// II. Try to match the json key with one in the tsv file
-						if ( headerFields.includes(columnLowercase) ) {
+                // 1. Style or unstyle table row
 
-							for ( let index = 0; index < this.columnToCategoryTable.length; index++ ) {
+                // A. Determine if category-column linking or unlinking has occurred
+                const linking = !( this.selectedCategory === this.columnToCategoryMap[p_clickData.column] );
```

Comment:
Could this just be 
```suggestion
                const linking = this.selectedCategory !== this.columnToCategoryMap[p_clickData.column] ;
```
? If so, I'd find that easier to read